tensorflow/java/src/main/java/org/tensorflow/EagerOperationBuilder.java

```diff
@@ -0,0 +1,249 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow;
+
+import java.nio.charset.Charset;
+
+/** 
+ * An {@link OperationBuilder} for building {@link Operation}s that are executed eagerly.
+ */
+final class EagerOperationBuilder implements OperationBuilder {
+
+  EagerOperationBuilder(EagerSession session, String type, String name) {
+    this.session = session;
+    this.type = type;
+    this.name = name;
+    this.nativeRef = new NativeReference(session, this, allocate(session.nativeHandle(), type));
+  }
+
+  @Override
+  public Operation build() {
+    // TODO (karllessard) Execute the eager operation and pass output tensor handles to new EagerOperation class
+    throw new UnsupportedOperationException("Eager execution is not supported yet");
+  }
+
+  @Override
+  public EagerOperationBuilder addInput(Output<?> input) {
+    addInput(nativeRef.opHandle, input.getUnsafeNativeHandle());
+    return this;
+  }
+
+  @Override
+  public OperationBuilder addInputList(Output<?>[] inputs) {
+    long[] inputHandles = new long[inputs.length];
+    for (int i = 0; i < inputs.length; ++i) {
+      inputHandles[i] = inputs[i].getUnsafeNativeHandle();
+    }
+    addInputList(nativeRef.opHandle, inputHandles);
+    return this;
+  }
+
+  @Override
+  public OperationBuilder addControlInput(Operation control) {
+    throw new UnsupportedOperationException("Control inputs are not supported in an eager execution environment");
+  }
+
+  @Override
+  public OperationBuilder setDevice(String device) {
+    setDevice(nativeRef.opHandle, device);
+    return this;
+  }
+
+  @Override
+  public OperationBuilder setAttr(String name, String value) {
+    return setAttr(name, value.getBytes(Charset.forName("UTF-8")));
+  }
+
+  @Override
+  public OperationBuilder setAttr(String name, String[] values) {
+    Charset utf8 = Charset.forName("UTF-8");
```

Comment:
`StandardCharsets.UTF_8`?

Reply:
Yep, that would fit too. The "UTF-8" literal is used in many other places in the client though, but I suggest we update those later as we go.